lib/config-generator.js

```diff
@@ -0,0 +1,253 @@
+                ]
+            },
+            {
+                type: "select",
+                name: "moduleType",
+                message: "What type of modules does your project use?",
+                initial: 0,
+                choices: [
+                    { message: "JavaScript modules (import/export)", name: "esm" },
+                    { message: "CommonJS (require/exports)", name: "commonjs" },
+                    { message: "None of these", name: "none" }
+                ]
+            },
+            {
+                type: "select",
+                name: "framework",
+                message: "Which framework does your project use?",
+                initial: 0,
+                choices: [
+                    { message: "React", name: "react" },
+                    { message: "Vue.js", name: "vue" },
+                    { message: "None of these", name: "none" }
+                ]
+            },
+            {
+                type: "select",
+                name: "language",
+                message: "Does your project use TypeScript?",
+                choices: [
+                    { message: "No", name: "javascript" },
+                    { message: "Yes", name: "typescript" }
+                ],
+                initial: 0
+            },
+            {
+                type: "multiselect",
+                name: "env",
+                message: "Where does your code run?",
+                hint: "(Press <space> to select, <a> to toggle all, <i> to invert selection)",
+                initial: 0,
+                choices: [
+                    { message: "Browser", name: "browser" },
+                    { message: "Node", name: "node" }
+                ]
+            }
+        ];
+
+        const answers = await enquirer.prompt(questions);
+
+        Object.assign(this.answers, answers);
+
+        if (answers.purpose === "style") {
+
+            // TODO: need to specify the package version - they may export flat configs in the future.
+            const jsStyleGuides = [
+                { message: "Airbnb: https://github.com/airbnb/javascript", name: "airbnb", value: { packageName: "eslint-config-airbnb", type: "eslintrc" } },
+                { message: "Standard: https://github.com/standard/standard", name: "standard", value: { packageName: "eslint-config-standard", type: "eslintrc" } },
+                { message: "XO: https://github.com/xojs/eslint-config-xo", name: "xo", value: { packageName: "eslint-config-xo", type: "eslintrc" } }
+            ];
+            const tsStyleGuides = [
+                { message: "Standard: https://github.com/standard/eslint-config-standard-with-typescript", name: "standard", value: { packageName: "eslint-config-standard-with-typescript", type: "eslintrc" } },
+                { message: "XO: https://github.com/xojs/eslint-config-xo-typescript", name: "xo", value: { packageName: ["eslint-config-xo", "eslint-config-xo-typescript"], type: "eslintrc" } }
```

Comment:
Can we move these to the top of the file? There doesn't appear to be a need for them to be inline to this method.